### DISCUSSION TITLE
Creating an order programmatically

### DISCUSSION THREAD

**Discussion by johnchiverton** [saleor/saleor#9783] [Q&A]
Apologies for the lack of linebreaks. Too tired. Can't think. Don't hate me.

I'm sure this is a basic syntax issue but, where I've managed to solve this in the past, for draftOrderCreate it's proving to be a real headache. Here's the code:

`
mutation
{
  draftOrderCreate (
    input:{
      user: "VXNlcjoz",
      userEmail: "someuser@somedomain.com",
      billingAddress: {
        firstName: "John",
        lastName: "Smith",
        companyName: "Lovely Shop",
        streetAddress1: "22 Acacia Avenue",
        city: "Glasgow",
        postalCode: "AA1 1AA",
        country: "GB",
        phone: "01234567890",
      }
      shippingAddress: {
        firstName: "John",
        lastName: "Smith",
        companyName: "Lovely Shop",
        streetAddress1: "22 Acacia Avenue",
        city: "Glasgow",
        postalCode: "AA1 1AA",
        country: "GB",
        phone: "01234567890",
      }
      lines: {
        quantity: 2,
        variantId: "UHJvZHVjdFZhcmlhbnQ6NDMx",
      }
    }
  )
}
`

This isn't a full mutation, just trying to hit the bare minimum needed without errors before adding to it. The error I'm getting is:

`
graphql.error.graphql_error.GraphQLError: Field 'draftOrderCreate' of type 'DraftOrderCreate' must have a selection of subfields. Did you mean 'draftOrderCreate { ... }'?
`

I'm sure there's a simple solution, but I've been mashing my head against this for hours now without any joy. I've gone tired-blind. Help.

**Comment by korycins** [comment]
Hey, I think that you missed to provide a subfields returned from the mutations. See an example below. We call mutation, and we request some fields of the related object (see order and errors ). 
```
mutation draftCreate(
        $user: ID, $discount: PositiveDecimal, $lines: [OrderLineCreateInput!],
        $shippingAddress: AddressInput, $billingAddress: AddressInput,
        $shippingMethod: ID, $voucher: ID, $customerNote: String, $channel: ID,
        $redirectUrl: String
        ) {
            draftOrderCreate(
                input: {user: $user, discount: $discount,
                lines: $lines, shippingAddress: $shippingAddress,
                billingAddress: $billingAddress,
                shippingMethod: $shippingMethod, voucher: $voucher,
                channelId: $channel,
                redirectUrl: $redirectUrl,
                customerNote: $customerNote}) {
                    errors {
                        field
                        code
                        variants
                        message
                        addressType
                    }
                    order {
                        discount {
                            amount
                        }
                        discountName
                        redirectUrl
                        lines {
                            productName
                            productSku
                            quantity
                        }
                        billingAddress{
                            city
                            streetAddress1
                            postalCode
                        }
                        shippingAddress{
                            city
                            streetAddress1
                            postalCode
                        }
                        status
                        voucher {
                            code
                        }
                        customerNote
                    }
                }
        }
```

**Comment by timuric** [comment]
@johnchiverton you could use dashboard UI as a reference for mutations while creating orders. GraphQL inspector might come in handy https://chrome.google.com/webstore/detail/graphql-network-inspector/ndlbedplllcgconngcnfmkadhokfaaln?hl=en-GB
